### PR TITLE
Change Configurator.Create to Return a Scheduler

### PR DESCRIPTION
**Pull request by ahmad-diaa**
**What type of PR is this?**
/kind cleanup

/sig scheduling

**What this PR does / why we need it**:
This PR makes `Configurator.Create` return a `Scheduler` instead of `factory.Config`. It is a step towards removing `factory.Config` and making the scheduler independent of it.

**Which issue(s) this PR fixes**:
Part of #72547

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -348,17 +352,14 @@ func New(client clientset.Interface,
...
 	default:
 		return nil, fmt.Errorf("unsupported algorithm source: %v", source)
 	}
 	// Additional tweaks to the config produced by the configurator.
-	config.Recorder = recorder
-	config.DisablePreemption = options.disablePreemption
-	config.StopEverything = stopCh
-
-	// Create the scheduler.
-	sched := NewFromConfig(config)
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go]: `NewFromConfig` had a call to `metrics.Register`

**Review by alculquicondor — COMMENTED** [review]
can we already get rid of `scheduler.Config` and `scheduler.NewFromConfig`?

**Comment by ahmad-diaa** [comment]
> can we already get rid of `scheduler.Config` and `scheduler.NewFromConfig`?

I think the only remaining usage is that one:
https://github.com/kubernetes/kubernetes/blob/a05efc6731a1acc29689fdbf268d5af71f7633ed/pkg/scheduler/eventhandlers_test.go#L108-L113

**Comment by alculquicondor** [comment]
> > can we already get rid of `scheduler.Config` and `scheduler.NewFromConfig`?
> 
> I think the only remaining usage is that one:
> https://github.com/kubernetes/kubernetes/blob/a05efc6731a1acc29689fdbf268d5af71f7633ed/pkg/scheduler/eventhandlers_test.go#L108-L113

Can you move it to directly create a Scheduler? If a test is the only user, then that test is not really testing anything useful :)

File: pkg/scheduler/eventhandlers_test.go [hunk 2]
```diff
@@ -105,13 +106,13 @@ func TestSkipPodUpdate(t *testing.T) {
 	}
 	for _, test := range table {
 		t.Run(test.name, func(t *testing.T) {
-			c := NewFromConfig(&Config{
+			metrics.Register()
```
Comment (alculquicondor) [inline pkg/scheduler/eventhandlers_test.go]: is this one (in the test) actually needed?

Comment (ahmad-diaa) [reply pkg/scheduler/eventhandlers_test.go]: I am not really sure, just added it to be safe. However when removing it, the test is still successful.

Comment (alculquicondor) [reply pkg/scheduler/eventhandlers_test.go]: metrics is something usually only makes sense during runtime. Unless you are testing the metrics themselves. It doesn't look like that is the case here.

Comment (ahmad-diaa) [reply pkg/scheduler/eventhandlers_test.go]: Okay, done!

**Comment by alculquicondor** [comment]
You can squash

**Review by draveness — CHANGES_REQUESTED** [review]
/hold 
can you squash the commits?

**Comment by ahmad-diaa** [comment]
>/hold
can you squash the commits?

@draveness already squashed, just kept the removal of `factory.Config` in a separate commit as I believe it's a significant change on its own and not directly related to the other commit.

**Comment by draveness** [comment]
/hold cancel

Thanks for the clarification

**Pull request merged**